Haskell/syntax_test_haskell.hs

```diff
@@ -145,6 +163,62 @@
    countTheBeforeVowel = undefined
 
 
+--KEYWORDS
+
+import qualified Data.Vector.Mutable as MutableVector
+-- ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ meta.import.haskell
+-- ^^^ keyword.control.import.haskell
+--     ^^^^^^^^^ keyword.control.import.haskell
+--               ^^^^^^^^^^^^^^^^^^^ support.other.module.haskell
+--                                   ^^ keyword.control.import.haskell
+--                                      ^^^^^^^^^^^^^ support.other.module.haskell
+import Data.List.Split (splitOn)
+-- ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ meta.import.haskell
+-- ^^^ keyword.control.import.haskell
+--     ^^^^^^^^^^^^^^^ support.other.module.haskell
+--                     ^^^^^^^^^ meta.declaration.exports.haskell
+--                     ^ punctuation.section.group.begin.haskell
+--                      ^^^^^^^ variable.function.haskell
+--                             ^ punctuation.section.group.end.haskell
+
+test =
+--   ^ keyword.operator.haskell
+    let x = 2 in x * y
+--  ^^^ keyword.declaration.variable.haskell
+--            ^^ keyword.control.context.haskell
+    where
+--  ^^^^^ keyword.control.context.haskell
+        y = 1
+--        ^ keyword.operator.haskell
+
+test a = case a of
+--       ^^^^ keyword.control.conditional.select.haskell
+--              ^^ keyword.control.conditional.select.haskell
+    Nothing -> 0
+    Just n -> if n > 0
+--            ^^ keyword.control.conditional.if.haskell
+        then n
+--      ^^^^ keyword.control.conditional.then.haskell
+        else 0
+--      ^^^^ keyword.control.conditional.else.haskell
+
+
+--MISC
+
+    (group)
+--  ^^^^^^^ meta.group.haskell
+--  ^ punctuation.section.group.begin.haskell
+--        ^ punctuation.section.group.end.haskell
+
+    [1,2]
+--  ^^^^^ meta.sequence.haskell
```

Comment:
I would say "list" rather than "sequence".

Reply:
Sequence is the undocumented standard for … sequential types, along with `meta.mapping` for mappings.

Reply:
Ah, gotcha.